### ISSUE TITLE
New Rule: An `async` function should have at least one `await` or return a Promise

### ISSUE THREAD

**Issue by brianmhunt** [eslint/eslint#16270]
### Rule details

Detect when an async function has no `await`

### Related ECMAScript feature

async/await

### What type of rule is this?

Warns about a potential problem

### Example code

```js
The following is almost always wrong:


async function aFn () { b() }

// should be 

async function aFn () { await b() }

// or 

async function aFn () { return b() }


where `b()` is a function call.

It would be most likely an omission for an `async` function doesn't have an `await` or return a `Promise`.
```

### Why should this rule be in the core instead of a plugin?

This seems like a fairly universal and consistent consideration.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by nzakas** [comment]
There is already a rule for this:
https://eslint.org/docs/latest/rules/require-await

We don't track whether or not a function returns a promise because we don't have the type information necessary to do so.

**Comment by brianmhunt** [comment]
Thanks @nzakas -- I appreciate you pointing me to this, and sorry I didn't see it when searching.  That rule certainly covers this, so thanks for responding and closing.

**Comment by ljharb** [comment]
It's also worth noting that `async function` has a ton of value even if it never uses `await` or returns a Promise, or even returns a *value*, which means the `require-await` rule (like this proposed one) is actually harmful when naively applied to a codebase. The `async` keyword guarantees, statically, that a function will never throw a synchronous exception, and will only ever return a Promise - this has value no matter what the function body contains.
